ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh

```diff
@@ -383,7 +370,18 @@ function check_upgrade_status() {
             echo -e "Upgrade stuck at updating RHEL worker, need to run the RHEL worker upgrade later...\n\n"
             return 0
         fi
-        dump_status_if_unexpected
+        # assuming oc to equal TARGET_MINOR_VERSION, log abnormal "upgrade status" if available (version 4.16+)
+        if [[ -n "${TARGET_MINOR_VERSION}" ]] && [[ "${TARGET_MINOR_VERSION}" -ge "16" ]] ; then
+            # capture "update health" section of status, ignoring the title and the last line
+            # whenever an error is present, "Run with --details=health" is the last line we should remove.
+            # with no error, it'll remove "Update is proceeding well" resulting in empty $stat that we simply ignore
+            stat="$(env OC_ENABLE_CMD_UPGRADE_STATUS=true oc adm upgrade status 2>&1 | sed -ne "/= Update Health =/,$ p" | sed '1d;2d;$d' || true)"
+            # if update health messages exist, and the mesage is different from previous (ignoring time difference using $filter), dump "Update Health" to log
+            if [ -n "$stat" ] && ! diff -qw <(sed -zE "s/${filter}//g" <<< "${stat}") <(sed -zE "s/${filter}//g" <<< "${oldstat}") >/dev/null ; then
+                echo "${stat}"
+                oldstat=${stat}
```

Comment:
In the [log](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/53915/rehearse-53915-periodic-ci-openshift-openshift-tests-private-release-4.17-arm64-stable-4.17-upgrade-from-stable-4.16-aws-ipi-proxy-sts-f28/1808142830237913088/artifacts/aws-ipi-proxy-sts-f28/cucushift-upgrade-toimage/build-log.txt), there is a shell syntax issue.
```
environment: line 381: oldstat: unbound variable
```

Reply:
good catch. it is basically unbound for the first use because it still haven't been set to $stat yet. I"m fixing this by assigning an initial value to it, shouldn't matter which as "the first time" we see any value in stat we can print it safely.